perfkitbenchmarker/linux_benchmarks/cloudsuite_websearch_benchmark.py

```diff
@@ -12,197 +12,129 @@
-  for vm in solr_nodes:
-    if vm == solr_nodes[0]:
-      solr.StartWithZookeeper(vm, SOLR_PORT, server_heap_size, False)
-    else:
-      solr.Start(vm, SOLR_PORT, solr_nodes[0], SOLR_PORT + 1000,
-                 server_heap_size, False)
-
-
-def _PrepareClient(clients):
-  """Prepares client machine in the benchmark."""
-  client_master = clients[0]
-  for client in clients:
-    client.Install('faban')
-    client.RemoteCommand('cd {0} && '
-                         'wget -O - {1} | '
-                         'tar -xzf -'.format(
-                             faban.FABAN_HOME_DIR, SEARCH_DRIVER_URL))
-    client.RemoteCommand('cd {0}/search && '
-                         'sed -i "/faban.home/c\\faban.home={0}" '
-                         'build.properties && '
-                         'sed -i "/ant.home/c\\ant.home='
-                         '{1}" build.properties && '
-                         'sed -i "/faban.url/c\\faban.url='
-                         'http://localhost:9980/" build.properties'.format(
-                             faban.FABAN_HOME_DIR, ANT_HOME_DIR))
-  faban.Start(client_master)
-  time.sleep(20)
+def _HasDocker(vm):
+  resp, _ = vm.RemoteCommand('command -v docker',
+                             ignore_failure=True,
+                             suppress_warning=True)
+  if resp.rstrip() == "":
+    return False
+  else:
+    return True
 
 
 def Prepare(benchmark_spec):
-  """Install cloudsuite web search and start the server on all machines.
+  """Install docker. Pull the required images from DockerHub.
+  Start Solr index node and client.
 
   Args:
     benchmark_spec: The benchmark specification. Contains all data that is
         required to run the benchmark.
   """
   vms = benchmark_spec.vms
-  servers = benchmark_spec.vm_groups['workers']
+  servers = benchmark_spec.vm_groups['servers'][0]
   clients = benchmark_spec.vm_groups['clients']
 
-  def PrepareVM(vm):
-    vm.Install('wget')
-    vm.RemoteCommand('mkdir -p {0}'.format(
-                     CLOUDSUITE_WEB_SEARCH_DIR))
+  for vm in vms:
+    if not _HasDocker(vm):
+      vm.Install('docker')
+
+  server_cmd = ('sudo echo \'DOCKER_OPTS="-g %s"\''
+                '| sudo tee /etc/default/docker > /dev/null' % (DISK_PATH))
+  stdout, _ = servers.RemoteCommand(server_cmd, should_log=True)
+
+  server_cmd = ('sudo service docker restart')
+  stdout, _ = servers.RemoteCommand(server_cmd, should_log=True)
+
+  for vm in clients:
+    vm.RemoteCommand('sudo docker pull cloudsuite/web-search:client')
 
-  vm_util.RunThreaded(PrepareVM, vms)
+  servers.RemoteCommand('sudo docker pull cloudsuite/web-search:server')
 
-  _PrepareSolr(servers)
-  _PrepareClient(clients)
-  _BuildIndex(servers)
+  server_cmd = ('sudo docker run -d --net host '
+                '--name server cloudsuite/web-search:server %s 1' %
+                (FLAGS.cs_websearch_server_heap_size))
+  stdout, _ = servers.RemoteCommand(server_cmd, should_log=True)
+
+  time.sleep(60)
```

Comment:
It gives the Solr server enough time to create the collection and stop. Without the sleep, it might be that the client starts before this server task is done, and hence thinks that the server has already loaded the index and starts running.